src/main/java/com/shipmentEvents/demo/TestBean2.java

```diff
@@ -0,0 +1,18 @@
+package com.shipmentEvents.handlers;
+
+import java.io.File;
+import java.io.FileWriter;
+
+public class TestBean2 {
+
+	public void doBusiness() {
+		
+		try {
+			File file = new File("c:\\temp\\test.txt");
+			FileWriter fw = new FileWriter(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fw*. Currently, there are execution paths that do not contain closure statements. Either a) close *fw* in a try-finally block or b) close the resource by declaring *fw* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).